lib/src/supabase.dart

```diff
@@ -90,23 +90,32 @@ class SupabaseClient {
     return rest.rpc(fn, params: params);
   }
 
-  /// Remove all subscriptions.
-  Future removeAllSubscriptions() async {
-    final subscriptions = getSubscriptions();
-    final futures = subscriptions.map((sub) => removeSubscription(sub));
-    await Future.wait(futures);
+  /// Closes and removes all subscriptions and returns a list of removed
+  /// subscriptions and their errors.
```

Comment:
It seems like we are not returning errors at this point. 
```suggestion
  /// Closes and removes all subscriptions and returns a list of removed subscriptions.
```

Reply:
i made a few changes. ```_closeChannel``` was renamed (like in js client) a now returns nullable ```SupabaseRealtimeError```. so we can return from removeSubscription ```RemoveSubscriptionResult``` with error and number of opened connections.

Reply:
Nice one!